interfaces/builtin/multipass_support.go

```diff
@@ -0,0 +1,156 @@
+# AppArmor.
+
+
+# Multipass has a server/client design, using a socket for IPC. The daemon runs
+# as root, but makes the socket accessible to anyone in the sudo group.
+# Need to permit chown and chgrp of the socket.
+capability chown,
+
+# Multipass uses gRPC for IPC, which calls shutdown on the socket
+unix ( shutdown) addr="@snap.@{SNAP_INSTANCE_NAME}.**",
+
+# Multipass generates AppArmor profiles for the utility processes it spawns.
+/sbin/apparmor_parser ixr,
+/etc/apparmor{,.d}/{,**} r,
+/etc/apparmor.d/{,cache/}multipass* rw,
+/sys/kernel/security/apparmor/{,**} r,
+/sys/kernel/security/apparmor/.remove w,
+/sys/kernel/security/apparmor/.replace w,
+
+# Allow running utility processes under the specialized AppArmor profiles.
+# These profiles will prevent utility processes escaping confinement.
+capability mac_admin,
+
+# Multipass will also use privilege separation when running utility processes
+capability setuid,
+capability setgid,
+
+# Some utility process (e.g. dnsmasq) will drop root privileges after startup and
+# change to another user. Multipass needs ability to stop them.
+capability kill,
+
+# Profiles Multipass generates have a naming scheme, restrict any profile changes to
+# those matching that scheme. Need unsafe to prevent env scrubbing.
+change_profile unsafe /var/snap/{@{SNAP_NAME},@{SNAP_INSTANCE_NAME}}/usr/bin/* -> multipass.*,
+signal (send) peer=multipass.*,
+ptrace (read, trace) peer=multipass.*,
+`
+
+const multipassSupportConnectedPlugSecComp = `
+# Description: allow operating as the Multipass daemon, and enable its various utilites
+# (qemu, qemu-img, dnsmasq, sshfs_server)
+
+# Note that this profile contains the union of all the necessary syscalls each of the
+# utilities requires. We rely on Multipass to generate specific AppArmor profiles
+# for each child process, to further restrict their abilities.
+
+# Multipass has server/client design, using a socket for IPC.
+# Note: Qemu has several features that also open sockets, but the AppArmor profile
+# Multipass will execute Qemu under locks down its filesystem access dramatically
+# and prevents it creating these sockets.
+accept
+accept4
+bind
+listen
+
+# dnsmasq fails unless it can drop supplementary groups
+setgroups 0 -
+setgroups32 0 -
+
+# Multipassd needs to chown its socket to a non-root user. sshfs_server also
+# will need these (see later).
+chown
+chown32
+
+# Multipassd will have a child process - sshfs_server - which allows mounting a
+# user-specified directory on the host into the VM. Here we permit typical filesytem
+# syscalls with the knowledge that Multipass will generate a specific AppArmor
+# profile for sshfs_server, restricting any filesystem access to just the
+# user-specified directory.
+
+# The default seccomp profile already includes these syscalls with the note:
+# "until we have syscall arg filtering (LP: #1446748)"
+# Am specifying them here as sshfs_server needs these syscalls unfiltered.
+chmod
+fchmod
+fchmodat
```

Comment:
These are already allowed in the default template.

Reply:
Right. I was paranoid as the default template has these listed with a precursor that reads like "they might be removed in future". I'll remove these and trust they won't :)